.github/workflows/pr.yaml

```diff
@@ -0,0 +1,46 @@
+name: Build only
+
+on:
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+env:
+  CANONICAL_DOMAIN: www.spinkube.dev
```

Comment:
nit: I don't think we need to set this same domain for just building, and we'll remove it when we work on previews... but then again it isn't hurting anything :)